File: server/datasets/python_source-code/Multiple-G1/renamed_46_bst.py
class Node_10:
    def __init__(self, val_10):
        self.val_10 = val_10
        self.lhs_10 = None
        self.rhs_10 = None

class Tree_10:
    def __init__(self):
        self.rt_10 = None

    def ins_10(self, val_10):
        if self.rt_10 is None:
            self.rt_10 = Node_10(val_10)
        else:
            self._ins_10_rec(self.rt_10, val_10)

    def _ins_10_rec(self, nd_10, val_10):
        if val_10 < nd_10.val_10:
            if nd_10.lhs_10 is None:
                nd_10.lhs_10 = Node_10(val_10)
            else:
                self._ins_10_rec(nd_10.lhs_10, val_10)
        elif val_10 > nd_10.val_10:
            if nd_10.rhs_10 is None:
                nd_10.rhs_10 = Node_10(val_10)
            else:
                self._ins_10_rec(nd_10.rhs_10, val_10)

    def srch_10(self, val_10):
        return self._srch_10_rec(self.rt_10, val_10)

    def _srch_10_rec(self, nd_10, val_10):
        if nd_10 is None or nd_10.val_10 == val_10:
            return nd_10
        if val_10 < nd_10.val_10:
            return self._srch_10_rec(nd_10.lhs_10, val_10)
        return self._srch_10_rec(nd_10.rhs_10, val_10)

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_g1/renamed_46_bst.java
class Node_10 {
    int val_10;
    Node_10 lhs_10, rhs_10;
    public Node_10(int item) {
        val_10 = item;
        lhs_10 = rhs_10 = null;
    }
}
public class Tree_10 {
    public static void main(String[] args) {
        Tree_10 bst = new Tree_10();
        bst.ins_10(50);
        bst.srch_10(50);
    }
    Node_10 rt_10;
    public void ins_10(int val_10) {
        rt_10 = in_rec_10(rt_10, val_10);
    }
    Node_10 in_rec_10(Node_10 rt_10, int val_10) {
        if (rt_10 == null) return new Node_10(val_10);
        if (val_10 < rt_10.val_10) rt_10.lhs_10 = in_rec_10(rt_10.lhs_10, val_10);
        else if (val_10 > rt_10.val_10) rt_10.rhs_10 = in_rec_10(rt_10.rhs_10, val_10);
        return rt_10;
    }
    Node_10 srch_10(int val_10) {
        return sr_rec_10(rt_10, val_10);
    }
    Node_10 sr_rec_10(Node_10 rt_10, int val_10) {
        if (rt_10 == null || rt_10.val_10 == val_10) return rt_10;
        if (rt_10.val_10 < val_10) return sr_rec_10(rt_10.rhs_10, val_10);
        return sr_rec_10(rt_10.lhs_10, val_10);
    }
}